packages/dcos-integration-test/extra/test_service_discovery.py

```diff
@@ -125,6 +128,206 @@ def _pool_for_mesos_dns():
             assert r_data['my_ip'] == service_points[0].host
 
 
+# There are several combinations we have to try of the way that Navstar DNS records are
+# generated:
+#
+# Containerizers:
+# -Mesos
+# -Docker
+#
+# Network type:
+# -Bridged
+# -Host
+# -Overlay
+#
+# Record type:
+# -Container IP
+# -Agent IP
+# -Auto IP
+#
+# More info can be found here: https://dcos.io/docs/1.8/usage/service-discovery/dns-overview/
+DNSAddresses = collections.namedtuple("DNSAddresses", ["container", "agent", "auto"])
+MarathonAddresses = collections.namedtuple("MarathonAddresses", ["host", "container"])
+
+
+def get_ipv4_addresses(hostname):
+    res = socket.getaddrinfo(hostname, 0, family=socket.AF_INET, type=socket.SOCK_STREAM)
+    return frozenset([sockaddr[0] for (family, type, proto, canonname, sockaddr) in res])
+
+
+def get_dns_addresses_by_app_name(app_name):
+    container_ip_name = '{}.marathon.containerip.dcos.thisdcos.directory'.format(app_name)
+    agent_ip_name = '{}.marathon.agentip.dcos.thisdcos.directory'.format(app_name)
+    auto_ip_name = '{}.marathon.autoip.dcos.thisdcos.directory'.format(app_name)
+    container_ips = get_ipv4_addresses(container_ip_name)
+    agent_ips = get_ipv4_addresses(agent_ip_name)
+    auto_ips = get_ipv4_addresses(auto_ip_name)
+    return DNSAddresses(container_ips, agent_ips, auto_ips)
+
+
+def get_marathon_addresses_by_service_points(service_points):
+    marathon_host_addrs = frozenset([point.host for point in service_points])
+    marathon_ip_addrs = frozenset([point.ip for point in service_points])
+    return MarathonAddresses(marathon_host_addrs, marathon_ip_addrs)
+
+
+def test_if_service_discovery_works_navstar_mesos_container_host_network(dcos_api_session):
```

Comment:
How about this name? test_if_service_discovery_mesos_host, test_if_service_discovery_mesos_overlay .... and so on